critical/back_end/modules/nat_instance/main.tf

```diff
@@ -0,0 +1,186 @@
+resource "aws_security_group" "nat_instance" {
+  vpc_id      = var.vpc_id
+  description = "Security group for NAT instance ${var.name}"
+
+  tags = {
+    Name = "nat-${var.name}"
+  }
+
+  egress {
+    cidr_blocks       = ["0.0.0.0/0"]
+    from_port         = 0
+    to_port           = 0
+    protocol          = "-1"
+  }
+
+  ingress {
+    cidr_blocks       = [var.cidr_block_private]
+    from_port         = 0
+    to_port           = 0
+    protocol          = "-1"
+  }
+}
+
+resource "aws_network_interface" "nat_instance" {
+  security_groups   = [aws_security_group.nat_instance.id]
+  subnet_id         = var.public_subnet
+  source_dest_check = false
+  description       = "ENI for NAT instance ${var.name}"
+
+  tags = {
+    Name = "nat-${var.name}"
+  }
+}
+
+resource "aws_eip" "nat_instance" {
+  network_interface = aws_network_interface.nat_instance.id
+
+  tags = {
+    Name = "nat-${var.name}"
+  }
+}
+
+resource "aws_route" "nat_instance" {
+  route_table_id         = var.private_subnet_route_table_id
+  destination_cidr_block = "0.0.0.0/0"
+  network_interface_id   = aws_network_interface.nat_instance.id
+}
+
+# AMI of the latest Amazon Linux 2
+data "aws_ami" "nat_instance" {
```

Comment:
How come we manually set up the NAT stuff on an AL2 AMI rather than using one of Amazon's NAT AMIs?

Reply:
That’s the way it was done in the module I copied, and I didn't want to change it more than necessary.